### PR TITLE
just moving code around. made !leaders.

### PR DESCRIPTION
**Pull request by byn3**

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -49,11 +54,10 @@ client.on('message', message => {
 	}
 
 	if (message.content === `${prefix}lt3` || message.content === `${prefix}roster`) {
+		// probably should just remove the `${prefix}lt3` above or change it to
```
Comment (driedyam) [inline bot.js]: I like this

File: bot.js [hunk 2]
```diff
@@ -5,8 +5,13 @@ const {MAPLE_STORY_CLASSES} = require("./constants.json")
 const client = new Discord.Client();
 
 /* importing functions from the commands dir */
-var add = require('./commands/add.js');
-var remove = require('./commands/remove.js');
+const add = require('./commands/add.js');
+const remove = require('./commands/remove.js');
+const postRoster = require('./commands/postRoster.js');
```
Comment (driedyam) [inline bot.js]: Is this postRoster instead of roster to avoid naming collisions with roster being used?

File: bot.js [hunk 3]
```diff
@@ -49,11 +54,10 @@ client.on('message', message => {
...
 
 	if (message.content === `${prefix}lt3` || message.content === `${prefix}roster`) {
+		// probably should just remove the `${prefix}lt3` above or change it to
+		// import from ./guilds/{filename} strip the .json for porability and customization for other guilds
 		let roster = getRoster()
-		let rosterMsg = ""
-		roster = roster.map(member => member.name).sort()
-		roster.forEach(member => rosterMsg += member + " " )
-		message.channel.send(`${rosterMsg} (${roster.length})`)
+		postRoster.postRoster(message, roster);
```
Comment (driedyam) [inline bot.js]: If we are going to extract all the commands into their own objects, what do you think about the paradigm of them all having a 'send' command, or 'render'? So that you would see:
```
postRoster.render(message, roster) 
find.render(message, roster)
demote.render(message, roster, leaders, name);
```

File: commands/promote.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+var fs = require("fs");
+
+module.exports = {
+  promoteCommand: function (message, roster, leaders) {
+    // if (leaders.length < 2) {
```
Comment (driedyam) [inline commands/promote.js]: Comments?

File: bot.js (hunk 2 above)
Comment (byn3) [reply bot.js]: kind of yea. Later on, I want to use a global roster variable since each function uses a let roster = getRoster() and having that global will speed the bot up a bunch.

TLDR: postRoster is to avoid naming collision of the future global variable we don't have yet.

File: bot.js (hunk 3 above)
Comment (byn3) [reply bot.js]: breaking things down is all fine. the decision is up to the tech lead and you. im literally just moving around code into their own files. Not really adding or changing anything at all unless i encounter a little bug.

Comment (driedyam) [reply bot.js]: Sounds like a job for @tommymurphy (Big T)

File: commands/promote.js (hunk 4 above)
Comment (byn3) [reply commands/promote.js]: Yea i don't know what y'all want to do with the promote function because it assumes 2 groups is the max and you cant promote some other member to run groups 3, 4, 5... What if theres a whale guild that can run more groups? then having a "there's already 2 groups" rejection kinda wont work for them. if users are manually promoting and demoting the exped leaders, there doesnt need to be a hard limit since they know the limit themselves. it might mess with group or balance logic though. I havent looked into those yet.

TLDR: just wanted confirmation from tom the bomb that this was the right decision to remove the max of 2 leader restriction message. nothing changed in the logic.

Comment (driedyam) [reply commands/promote.js]: In the groups rewrite I'm only touching !groups and balance, but I think the removal of this restriction makes sense. It should be an arbitrary amount of leaders and the app should just do the right thing

File: bot.js [hunk 5]
```diff
@@ -5,8 +5,13 @@ const {MAPLE_STORY_CLASSES} = require("./constants.json")
 const client = new Discord.Client();
 
 /* importing functions from the commands dir */
-var add = require('./commands/add.js');
-var remove = require('./commands/remove.js');
+const add = require('./commands/add.js');
```
Comment (tommymurphy) [inline bot.js]: To make this cleaner: `require('/commands/add.js').add;` 
Then you don't get that weird add.add function call

File: bot.js [hunk 6]
```diff
@@ -84,52 +88,35 @@ client.on('message', message => {
...
-		        return;
-		    };
-		    message.channel.send(`Promoted ${name} to expedition leader!`)
-		});
+		let roster = getRoster();
+		promote.promoteCommand(message, roster, leaders);
+	}
+
+	if (message.content.substring(0,8) === `${prefix}leaders` && isAdmin) {
+		let roster = getRoster();
```
Comment (tommymurphy) [inline bot.js]: This can be done in one line:` message.channel.send(getRoster.filter(member => member.leader).join(" ,"))`

File: bot.js [hunk 7]
```diff
@@ -84,52 +88,35 @@ client.on('message', message => {
...
-		demoted.leader = false
-		leaders = leaders.filter(leader => leader.name.toLowerCase() !== name.toLowerCase())
-		fs.writeFile("./guilds/lt3.json", JSON.stringify({"lt3":roster}, null, 4), (err) => {
-				if (err) {
-						console.error(err);
-						return;
-				};
-				message.channel.send(`Removed ${name} from expedition leaders!`)
-		});
+		demote.demoteCommand(message, roster, leaders, name);
```
Comment (tommymurphy) [inline bot.js]: Change naming convention as seen in first comment at top of file so you can just do `demote(args)`

File: commands/promote.js (hunk 4 above)
Comment (tommymurphy) [reply commands/promote.js]: Remove the comments for now and leave as is until Chris/me refactors for multiple groups imo

File: bot.js (hunk 3 above)
Comment (tommymurphy) [reply bot.js]: @driedyam Tried this at one point and had issues.. but worth another shot

File: commands/promote.js (hunk 4 above)
Comment (byn3) [reply commands/promote.js]: I'm adding some null and more than 1 input checks if that's okay.
